### PR TITLE
PKCS11Module::SetOwner fix for tee identity

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: include/aos/common/crypto.hpp [hunk 1]
```diff
@@ -595,6 +606,15 @@ class ProviderItf {
      */
     virtual Error ASN1DecodeOID(const Array<uint8_t>& inOID, Array<uint8_t>& dst) = 0;
 
+    /**
+     * Creates SHA1 UUID(version 5) based on a given name space and data.
+     *
+     * @param namespace name space.
```
Comment (mlohvynenko) [inline include/aos/common/crypto.hpp:612]: there's no 'namespace' param. Please make it consistent.

File: include/aos/common/crypto/mbedtls/cryptoprovider.hpp [hunk 2]
```diff
@@ -165,6 +165,15 @@ class MbedTLSCryptoProvider : public x509::ProviderItf {
      */
     Error ASN1DecodeOID(const Array<uint8_t>& inOID, Array<uint8_t>& dst) override;
 
+    /**
+     * Creates SHA1 hash based on a given name space and data.
+     *
+     * @param namespace name space.
```
Comment (mlohvynenko) [inline include/aos/common/crypto/mbedtls/cryptoprovider.hpp:171]: there's no 'namespace' param. Please make it consistent.

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 3]
```diff
@@ -641,6 +642,37 @@ Error MbedTLSCryptoProvider::ASN1DecodeOID(const Array<uint8_t>& inOID, Array<ui
     return crypto::ASN1RemoveTag(inOID, dst, MBEDTLS_ASN1_OID);
 }
 
+RetWithError<uuid::UUID> MbedTLSCryptoProvider::CreateSHA1(const uuid::UUID& space, const String& data)
+{
+    constexpr auto version = 5;
```
Comment (mlohvynenko) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:647]: shouldn't it be a static const in the classes's private section?

File: src/common/crypto/mbedtls/cryptoprovider.cpp [hunk 4]
```diff
@@ -641,6 +642,37 @@ Error MbedTLSCryptoProvider::ASN1DecodeOID(const Array<uint8_t>& inOID, Array<ui
...
+
+    int ret = mbedtls_sha1(buffer.Get(), buffer.Size(), sha1.Get());
+    if (ret != 0) {
+        return {{}, AOS_ERROR_WRAP(ret)};
+    }
+
+    // copy lowest 16 bytes
+    uuid::UUID result = Array<uint8_t>(sha1.Get(), uuid::cUUIDLen);
+
+    result[6] = (result[6] & 0x0f) | uint8_t((version & 0xf) << 4);
```
Comment (mlohvynenko) [inline src/common/crypto/mbedtls/cryptoprovider.cpp:670]: can you please add more specific comments that explains this magic?

File: src/common/pkcs11/pkcs11.cpp [hunk 5]
```diff
@@ -238,17 +238,15 @@ Array<uint8_t> ConvertToAttributeValue(T& val)
 
 Error GenPIN(String& pin)
 {
-    if (pin.MaxSize() == 0) {
-        return ErrorEnum::eNone;
-    }
+    const auto PinLength = Min<size_t>(cGenPINLen, pin.MaxSize());
```
Comment (mlohvynenko) [inline src/common/pkcs11/pkcs11.cpp:241]: Shouldn't it be a lower camel case variable?

File: include/aos/common/crypto.hpp (hunk 1 above)
Comment (mykola-kobets-epam) [reply include/aos/common/crypto.hpp]: done

File: include/aos/common/crypto/mbedtls/cryptoprovider.hpp (hunk 2 above)
Comment (mykola-kobets-epam) [reply include/aos/common/crypto/mbedtls/cryptoprovider.hpp]: done

File: src/common/crypto/mbedtls/cryptoprovider.cpp (hunk 3 above)
Comment (mykola-kobets-epam) [reply src/common/crypto/mbedtls/cryptoprovider.cpp]: I don't know, it is used for sha1 specific case only.

File: src/common/crypto/mbedtls/cryptoprovider.cpp (hunk 4 above)
Comment (mykola-kobets-epam) [reply src/common/crypto/mbedtls/cryptoprovider.cpp]: done

File: src/common/pkcs11/pkcs11.cpp (hunk 5 above)
Comment (mykola-kobets-epam) [reply src/common/pkcs11/pkcs11.cpp:241]: For variable yes, for local const values probably we should use upper camel.

Comment (al1img) [reply src/common/pkcs11/pkcs11.cpp:241]: we use prefix `c` for const: `cPinLength`

File: include/aos/common/crypto.hpp [hunk 6]
```diff
@@ -595,6 +606,15 @@ class ProviderItf {
...
     virtual Error ASN1DecodeOID(const Array<uint8_t>& inOID, Array<uint8_t>& dst) = 0;
 
+    /**
+     * Creates SHA1 UUID(version 5) based on a given name space and data.
+     *
+     * @param space name space.
+     * @param data input data.
+     * @result RetWithError<UUID>.
+     */
+    virtual RetWithError<uuid::UUID> CreateSHA1(const uuid::UUID& space, const String& data) = 0;
```
Comment (al1img) [inline include/aos/common/crypto.hpp:616]: data should be aos::Array<uint8_t>

Comment (mykola-kobets-epam) [reply include/aos/common/crypto.hpp]: fixed

File: src/common/crypto/mbedtls/cryptoprovider.cpp (hunk 3 above)
Comment (mykola-kobets-epam) [reply src/common/crypto/mbedtls/cryptoprovider.cpp]: renamed to cUUIDVersion

File: src/common/pkcs11/pkcs11.cpp (hunk 5 above)
Comment (mykola-kobets-epam) [reply src/common/pkcs11/pkcs11.cpp:241]: fixed

**Review by al1img — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Pull request merged**